cpp/src/lance/io/exec/project.cc

```diff
@@ -31,16 +32,23 @@ Project::Project(std::unique_ptr<ExecNode> child) : child_(std::move(child)) {}
 
 ::arrow::Result<std::unique_ptr<Project>> Project::Make(
     std::shared_ptr<FileReader> reader,
-    std::shared_ptr<::arrow::dataset::ScanOptions> scan_options,
-    std::optional<int32_t> limit,
-    int32_t offset) {
+    std::shared_ptr<::arrow::dataset::ScanOptions> scan_options) {
   auto& schema = reader->schema();
   auto projected_arrow_schema = scan_options->projected_schema;
   if (projected_arrow_schema->num_fields() == 0) {
     projected_arrow_schema = scan_options->dataset_schema;
   }
   ARROW_ASSIGN_OR_RAISE(auto projected_schema, schema.Project(*projected_arrow_schema));
 
+  std::optional<int64_t> limit;
+  int64_t offset;
+  if (scan_options->fragment_scan_options &&
+      scan_options->fragment_scan_options->type_name() == "lance") {
```

Comment:
Let's extract the string check into a util function?

Reply:
done